src/hooks/useAdTimeDetector.jsx

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useRef } from "react";
+
+const isAdTimeNow = (minutes, seconds) => {
+  const isPreThirtyAdTime =
+    (minutes === 27 && seconds >= 30) ||
+    minutes === 28 ||
+    minutes === 29 ||
+    (minutes === 30 && seconds === 0);
+
+  const isPreHourAdTime =
+    minutes === 57 ||
+    minutes === 58 ||
+    minutes === 59 ||
+    (minutes === 0 && seconds === 0);
+
+  return isPreThirtyAdTime || isPreHourAdTime;
+};
+
+const useAdTimeDetector = (handleChannelSwitch) => {
+  const isAdTimeAlreadyHandled = useRef(false);
+
+  useEffect(() => {
+    const checkAdTime = () => {
+      const now = new Date();
+      const minutes = now.getMinutes();
+      const seconds = now.getSeconds();
```

Comment:
```suggestion
      const [minutes, seconds] = [now.getMinutes(), now.getSeconds()];
```

구조 분해 할당을 이용하여 이렇게 한 번에 받아주는 코드는 어떤가요?